### PR TITLE
Add collection translator, fixes projecthydra/sufia#2142

### PR DESCRIPTION
**Pull request by hackmastera**
@cam156

### CODE REVIEW

File: lib/sufia/import/collection_translator.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  # Imports a Sufia 6.0-exported Collection into a Sufia PCDM Collection
+  class CollectionTranslator
+    def initialize(settings)
+      @collection_builder = CollectionBuilder.new(settings)
+    end
+
+    # loop through the files found in import_dir and map each Collection
+    # @param import_dir a directory containing export files, e.g.: collection_[id].json
+    # @return list of imported files
+    def import(import_dir, filename_prefix)
```
Comment (cam156) [inline lib/sufia/import/collection_translator.rb]: @hackmastera This code for opening files seems quite simmilar to the GenericFileTranslator: https://github.com/projecthydra/sufia/blob/7.2-migration/lib/sufia/import/generic_file_translator.rb#L17-L23

May indicate that a base class would be nice.

File: lib/sufia/import/collection_translator.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      files = Dir.glob(File.join(import_dir, '*')).select { |f| exported_json?(f, filename_prefix) }
+      files.each { |f| import_file(f) }
+    end
+
+    private
+
+      def import_file(filename)
+        Rails.logger.debug "Importing #{File.basename(filename)}"
+        file_str = File.read(filename)
+        json = JSON.parse(file_str)
```
Comment (cam156) [inline lib/sufia/import/collection_translator.rb]: This code is also strikingly similar to GenericFileTranslator.

File: lib/sufia/import/collection_translator.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        collection = @collection_builder.build(json)
+        collection.save
+      end
+
+      def exported_json?(f, filename_prefix)
```
Comment (cam156) [inline lib/sufia/import/collection_translator.rb]: This method is the same as the one in GFT except for the name.

**Comment by hackmastera** [comment]
@cam156 yes that's exactly the refactoring I was referring to earlier! I'm working on the config file bit first since it will get rid of the need for the username / password settings and simplify things, then will do this refactor. I think it would be easier for me to merge first if that's your only concern. But I defer to your judgment!

**Pull request merged**